### PR TITLE
Add projectsByOpenshift API Endpoint

### PR DESCRIPTION
**Pull request by cdchris12**
# Checklist

- [ ] Documentation has been written/updated
- [X] PR title is ready for changelog and subsystem label(s) applied

This PR adds the `projectsByOpenshift` API endpoint, which will show all projects which are hosted on a specified openshift.

### CODE REVIEW

File: services/api/src/resources/project/resolvers.ts [hunk 1]
```diff
@@ -187,6 +187,47 @@ export const getProjectByName: ResolverFn = async (
...
+) => {
+  let where;
+  try {
+    await hasPermission('project', 'viewAll');
+  } catch (err) {
+    if (!keycloakGrant) {
+      logger.warn('No grant available for getProjectByOpenshift');
+      return [];
+    }
+  }
```
Comment (rocketeerbkw) [inline services/api/src/resources/project/resolvers.ts]: This is not a sufficient way to verify the user is allowed to view the projects being returned. If the user has `viewAll` permissions, then we don't have to limit the SQL query. In the `else` case, we have to figure out what projects the user _does_ have access to, and limit the SQL query to only return those projects.

See [`getAllProjects`](https://github.com/amazeeio/lagoon/blob/main/services/api/src/resources/project/resolvers.ts#L50-L57) for an example.

**Review by rocketeerbkw — CHANGES_REQUESTED** [review]

File: services/api/src/resources/project/resolvers.ts (hunk 1 above)
Comment (cdchris12) [reply services/api/src/resources/project/resolvers.ts]: Thanks for the quick review. I've updated the PR accordingly. :)

**Comment by rocketeerbkw** [comment]
There probably needs to be some thought put into what `openshift` permissions are required for this query. There are `openshift:view` and `openshift:viewAll` permissions used in other places and the dependency is that the system needs to know the project first.

In other words, to know if a user has permission to view a specific openshift data, we must first figure out of that user is part of a project that has that openshift linked to it. I don't think this PR addresses that use case either.

If this is only for system engineers maybe it's not worth making available for non-admins to run. You could replace all the permissions try/catch stuff with two checks: `await hasPermission('openshift', 'viewAll');` and `await hasPermission('project', 'viewAll');` to limit the whole query.

**Comment by cdchris12** [comment]
@rocketeerbkw Thanks again for the feedback. :) 

With the latest push, I've added functionality to filter shown projects based on openshift permissions which are derived from the list of projects the user has access to. Hopefully this is what you were looking for here?

**Comment by rocketeerbkw** [comment]
Superseded by #2686

**Pull request closed without merging**